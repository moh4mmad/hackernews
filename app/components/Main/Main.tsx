export default function Main({ children }: { children: React.ReactNode }) {
  return <div className="px-8 py-8">{children}</div>;
}
