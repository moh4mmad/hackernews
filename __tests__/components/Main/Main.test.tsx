import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Main from '@/app/components/Main/Main';
import renderer from 'react-test-renderer';

describe('Page Main', () => {
  it('should render Main component', () => {
    const { container } = render(<Main />);
    expect(container.getElementsByClassName('px-8 py-8').length).toBe(1);
    expect(container).toMatchSnapshot();
  });
});
