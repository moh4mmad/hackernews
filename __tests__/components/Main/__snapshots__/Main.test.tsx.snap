// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Main should render Main component 1`] = `
<div>
  <div
    class="px-8 py-8"
  />
</div>
`;

exports[`Page Main snapshot should snapshot Main component 1`] = `
<div
  className="px-8 py-8"
/>
`;
